### DISCUSSION TITLE
Git organization of the project

### DISCUSSION THREAD

**Discussion by Tonterias** [Tonterias/SpingularJhipsterV7#6] [Q&A]
Let's agree here about how organize the Git structure and other issues related to Git!

**Comment by cchunduri** [comment]
Hi Mike, 

Here is the structure that I'm proposing. 

At first, after we generate the initial project via Jhipster(here we'll not keep complete JDL, we keep basic JDL which is needed for project generation) we'll push the project to the master branch. From this point onwards we'll be creating a branch for each feature. For example

- We're working on the managing posts(creating a post, updating a post, deleting post) for the CMS, then we'll create a branch saying **post_management** once the development is over, testing is also over and when developer gains confidence then we'll merge it to **master** branch and deletes the **post_management**. 

Also, we maintain one release branch, for example

- For the first version we planned user management, post-management. Once these two features completed, when we gained the confidence to give release then we create a branch on master saying the name **v1.0**. This branch will be there till the next version **v1.1** gets released. 
- If anyone wants to test upcoming features they'll checkout master and can test it. 

Please feel free to ask anything that comes to mind. Always welcome for the changes. 

Regards,
Chaitanya Chunduri

**Reply by Tonterias** [reply]
It is ok with me. With your structure, we can work on parts independently and everybody will see how we solved each part and we can explain things relating it with the commits that we made into master.

Thanks Chaitanya!

**Reply by Goldilakhmani** [reply]
Hi Mike/Chaitanya,

                 I am ok with Chaitanya suggestion. We can work on git branching strategy, Let us also  plan to come up with the design and post design closure we can kick off the development.

Thanks 
Goldi

**Reply by cchunduri** [reply]
> Hi Mike/Chaitanya,
> 
>                  I am ok with Chaitanya suggestion. We can work on git branching strategy, Let us also  plan to come up with the design and post design closure we can kick off the development.
> 
> Thanks 
> Goldi

Goldi,

Please take a look at the issue #3 for the design. Please feel free to ask any questions.

**Reply by Goldilakhmani** [reply]
Hi @Tonterias / @cchunduri  ,

Thanks for taking initiative, I will look into JDL and try to understand design. Is there any way to add notification so that i can receive email whenever there are new discussion regarding this. So I can pro-actively start contributing the same.

Regards
Goldi

**Reply by Tonterias** [reply]
Hi Goldi & Chaitanya,

I have tested version 7 and I think that we may have problems if we do it that way. Let's say that we make changes in the Post entities and we need those changes in another entity to make it work. 

Can we use those changes until the Post entity is merge to the main branch? I do not think so, please let me know.

If we can not, we may use another strategy as just using one development branch, right?

Take care

**Comment by mgarayarc** [comment]
Hi everybody,

I was thinking about a clean way to organize tasks and pendings and I found this method based on a task board such Scrum board. What do you think? Take a look:

https://www.youtube.com/watch?v=jhtbhSpV5YA&ab_channel=AkoDev

Regards!!
Miguel

**Reply by Tonterias** [reply]
Hi Miguel,

Great video. I think that is what @cchunduri was thinking of in the first comment. 

By the way, it talks about Projects which we should also use as [they are explained also here](https://www.youtube.com/watch?v=ff5cBkPg-bQ) 

I have created a few project tasks in the [dev branch that you can see here](https://github.com/Tonterias/SpingularJhipsterV7/projects/1?add_cards_query=is%3Aopen) which are the commits that I have already done so far to test it.

Well done.

**Comment by Goldilakhmani** [comment]
Hi all


Hope everyone is doing well, lets sync up sometime next week and start
planning the activities .

Thanks Miguel for joining us

Regards
Goldi

**Reply by Tonterias** [reply]
Hi Goldi,

Yeah we can try to do it over Hang-Outs chat sometime (or Meet for videoconference), but may be it becomes really difficult since Miguel is America's timezone, I'm in Europe's timezone, and you are both in Indias timezone (which leave us with a really narrow chance to find a common time). I will ask on HangOut. Check it out there, ok?

In the meantime, we should consider trying to use GitHub tools to work collaboratively and talk about each separate topic, like [entities here:]( https://github.com/Tonterias/SpingularJhipsterV7/discussions/3) and the other [MORE IMPORTANT general things here:](https://github.com/Tonterias/SpingularJhipsterV7/discussions/4).

I'm looking at the free [minibook from Jhipster by Matt Raible: ](https://www.infoq.com/minibooks/jhipster-mini-book-5/)  to get an idea about the order of things.
